GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/GVREmitter.java

```diff
@@ -0,0 +1,349 @@
+     * @param height volume height (along y-axis)
+     * @param depth volume depth (along z-axis)
+     */
+
+    public void setParticleVolume(float width, float height, float depth)
+    {
+        Vector3f center = new Vector3f(this.getTransform().getPositionX(),
+                this.getTransform().getPositionY(), this.getTransform().getPositionZ());
+
+        particleBoundingVolume = new float[]{center.x - width/2, center.y - height/2, center.z - depth/2,
+                                       center.x - width/2, center.y - height/2, center.z + depth/2,
+                                       center.x + width/2, center.y - height/2, center.z + depth/2,
+                                       center.x + width/2, center.y - height/2, center.z - depth/2,
+
+                                       center.x - width/2, center.y + height/2, center.z - depth/2,
+                                       center.x - width/2, center.y + height/2, center.z + depth/2,
+                                       center.x + width/2, center.y + height/2, center.z + depth/2,
+                                       center.x - width/2, center.y + height/2, center.z - depth/2};
+
+        BVSpawnTimes = new float[]{100, 0, 100, 0, 100, 0, 100, 0, 100, 0, 100, 0, 100, 0, 100, 0};
+
+        BVVelocities = new float[24];
+        for ( int i = 0; i < 24; i ++ )
+            BVVelocities[i] = 0;
+    }
+
+    protected void tickClock(float time)
+    {
+        currTime = time;
+
+        for (GVRSceneObject partObj: mActiveParticleMeshes)
+        {
+            partObj.getRenderData().getMaterial().setFloat("u_time", time);
+        }
+    }
+
+    /**
+     * @param emitRate The rate( #particles/second ) at which this emitter emits particles.
+     *                 Currently clamped to 600 particles/second.
+     */
+    public void setEmitRate(int emitRate)
+    {
+        if ( emitRate > MAX_EMIT_RATE)
+            mEmitRate = MAX_EMIT_RATE;
+        else
+            mEmitRate = emitRate;
+    }
+
+    /**
+     * The rate ( #particles/second ) at which this emitter is emitting
+     * @return
+     */
+    public int getEmitRate()
+    {
+        return mEmitRate;
+    }
+
+    /**
+     * The ( emit rate * age ) decide the total number of particles in the scene.
+     * @param age The time (in seconds) the particles from this emitter will remain active in the
+     *            scene.
+     */
+    public void setParticleAge ( float age )
+    {
+        mMaxAge = age;
+    }
+
+    /**
+     * @param size the initial particle size.
+     */
+    public void setParticleSize ( float size )
+    {
+        mParticleSize = size;
+    }
+
+    /**
+     * The range of velocities that a particle generated from this emitter can have.
+     * @param minV Minimum velocity that a particle can have
+     * @param maxV Maximum velocity that a particle can have
+     */
+    public void setVelocityRange( float minV, float maxV )
```

Comment:
There is no getter so you could make it more thread-safe by running the body on the gl thread.